src/main/java/io/codelex/flightplanner/adminapi/AdminController.java

```diff
@@ -0,0 +1,43 @@
+package io.codelex.flightplanner.adminapi;
+
+import io.codelex.flightplanner.flights.Flight;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/admin-api")
+public class AdminController {
+
+    private AdminService adminService;
+
+    public AdminController(AdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    @PutMapping("/flights")
+    @ResponseStatus(HttpStatus.CREATED)
+    public synchronized Flight addFlight(@Valid @RequestBody AddFlightRequest addFlightRequest) {
+        checkIfSameAirport(addFlightRequest);
+        return adminService.addFlight(addFlightRequest);
+    }
+
+    @DeleteMapping("/flights/{id}")
+    public synchronized void deleteFlight(@PathVariable("id") Long id) {
+        adminService.deleteFlight(id);
+    }
+
+    @GetMapping("/flights/{id}")
+    public Flight getFlight(@PathVariable("id") Long id) {
+        return adminService.fetchFlight(id);
+    }
+
+    private void checkIfSameAirport(AddFlightRequest addFlightRequest) {
```

Comment:
Pluss par to, ka validāciju veici tikko jau ir pieejami nepieciešamie dati, lai to izdarītu, jebšu iekš controller.